src/harness/reference_models/interference/aggregate_interference_test.py

```diff
@@ -0,0 +1,110 @@
+#    Copyright 2018 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+
+import os
+import unittest
+import json
+from reference_models.examples import testutils
+from reference_models.interference import aggregate_interference
+from reference_models.interference import interference as interf
+
+
+TEST_DIR = os.path.join(os.path.dirname(__file__), 'test_data')
+
+
+class TestAggregateInterference(unittest.TestCase):
+
+  @classmethod
+  def setUpClass(cls):
+
+    # Load the list of CBSD records with registration and grant details 
+    cbsd_filename = ['cbsd_uut_ut.json', 'cbsd_th1_ut.json']
+    cls.cbsd_list = []
+    for cbsd_file in cbsd_filename:
+      cbsd_record = json.load(open(os.path.join(TEST_DIR, cbsd_file)))
+      cls.cbsd_list.append(cbsd_record)
+
+    # Load the protection entity data with overlapping frequency of CBSD grants 
+    cls.fss_record = json.load(open(os.path.join(TEST_DIR, 'fss_ut.json')))
+    cls.ppa_record = json.load(open(os.path.join(TEST_DIR, 'ppa_ut.json')))
+    cls.pal_record = json.load(open(os.path.join(TEST_DIR, 'pal_ut.json')))
+    cls.gwpz_record = json.load(open(os.path.join(TEST_DIR, 'gwpz_ut.json'))) 
+    cls.esc_record = json.load(open(os.path.join(TEST_DIR, 'esc_ut.json')))
+
+
+  def setUp(self):
+    self.original_interference = interf.computeInterference
+
+  def tearDown(self):
+    interf.computeInterference = self.original_interference
+
+  def test_AggregateInterferenceFssCochannel(self):
+    expected_interference = {34.4116: {-100.57114: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\
```

Comment:
Did you check by hand these values are expected, or are they the results of a first run ?

Of course if you pass a specific propag_predictor, with L1 distance (abs(lat2-lat1) + abs(lon2-lon1)), easy to know what is the expected pathloss returned, and then verify by hand/excel that the calculation is ok.

Same comment about below tests about making sure that the reference has been validated/computed by hand or in excel sheet, as it now tells you if there is an issue with the current code. The stored reference then provides regression testing for future update.

Reply:
Have   changed the distance type to L1, calculated the path loss value manually and   verified the same with the executed results. Results are available up on request.